core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java

```diff
@@ -0,0 +1,62 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class UseHashMapToConvertPhoneNumberInWordsToNumber {
+    public static String convertPhoneNumberInWordToNumber(String phoneNumberInWord) {
+
+        Map<String, Integer> mulpliers = getWordAsMultiplier();
+        Map<String, String> digits = getWordAsDigit();
+
+        StringBuilder output = new StringBuilder();
+        Integer currentMultiplier = null;
+        String[] words = phoneNumberInWord.split(" ");
+
+        for (String word : words) {
+            Integer multiplier = mulpliers.get(word);
+            if (multiplier != null) {
+                if (currentMultiplier != null) {
+                    throw new IllegalArgumentException("Cannot have consecutive multipliers, at: " + word);
+                }
+                currentMultiplier = multiplier;
+            } else {
+                String digit = digits.get(word);
+                if (digit != null) {
+                    output.append(digit.repeat(currentMultiplier != null ? currentMultiplier : 1));
+                    currentMultiplier = null;
+                } else {
+                    throw new IllegalArgumentException("Invalid word: " + word);
+                }
+            }
+        }
+
+        return output.toString();
+    }
+
+    public static Map<String, Integer> getWordAsMultiplier() {
+        Map<String, Integer> multiplier = new HashMap<>();
+
+        multiplier.put("double", 2);
+        multiplier.put("triple", 3);
+        multiplier.put("quadruple", 4);
+
+        return multiplier;
+    }
+
+
+    public static Map<String, String> getWordAsDigit() {
```

Comment:
We don't need a function that continously builds this map. We need a static constant in the class containing this map. You can do that using `private static` fields and the `static { ... construct here ... }` constructor method.

In higher versions of Java, we can also build a map in an initialiser list using `Map.of`. Or we can use map building code from Guava if we really wanted to.

The map needs to be wrapped by `unmodifiableMap` https://www.geeksforgeeks.org/collections-unmodifiablemap-method-in-java-with-examples/ to make it read only.

Having a method repeatedly make a map is a bad example to set to the reader.

Reply:
Used the `private static Map.of()` method to fix the issue.